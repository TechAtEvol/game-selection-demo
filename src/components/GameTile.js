import React from 'react';
import { LazyLoadImage } from 'react-lazy-load-image-component';

const GameTile = props => {
  // TODO: Add a small snapshot test perhaps of the tile rendering
  // TODO: Better CSS for mobile view, use the full width
  const { game } = props;
  return (
    <div className="ui card">
      <div className = "image">
        <LazyLoadImage
          src = {game.thumbnailUrl}
          alt = {game.friendlyName}
        />
      </div>
      <div className="content">
        <div className="header">{game.name}</div>
        <div className="description">{game.description}</div>
      </div>
      <div className="extra content">
        <a href="http://mrgreen.se">
          <i className="play icon"></i>
          Spela h√§r
        </a>
      </div>
    </div>
  )
};
export default GameTile