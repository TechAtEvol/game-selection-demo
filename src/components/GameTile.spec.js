import React from 'react';
import renderer from 'react-test-renderer';
import GameTile from './GameTile'
import allGames from './../test-resources/all-games'

describe('GameTile', () => {
  const testGame = allGames[0];
  it('renders correctly', () => {
    const dom = renderer.create(
      <GameTile game={testGame}/>
    ).toJSON()
    expect(dom).toMatchSnapshot();
  })
});