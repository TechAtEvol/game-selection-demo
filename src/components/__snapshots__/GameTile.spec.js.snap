// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameTile renders correctly 1`] = `
<div
  className="ui card"
>
  <div
    className="image"
  >
    <span
      className=""
      style={
        Object {
          "display": "inline-block",
          "height": 0,
          "width": 0,
        }
      }
    />
  </div>
  <div
    className="content"
  >
    <div
      className="header"
    >
      1 Can 2 Can
    </div>
    <div
      className="description"
    >
      Jag kan, du kan... Tukan! Ta en promenad på den vilda sidan och besök Tukanen och hans vänners värld.  Detta orörda djungelparadis är fylld med färger, roliga frukter, vänliga djur och en Tukan som kommer driva ditt spel vilt med spänning.
    </div>
  </div>
  <div
    className="extra content"
  >
    <a
      href="http://mrgreen.se"
    >
      <i
        className="play icon"
      />
      Spela här
    </a>
  </div>
</div>
`;
